imgparse/imgparse.py

```diff
@@ -230,27 +233,65 @@ def parse_session_alt(image_path):
     return float(session_alt)
 
 
+def _get_terrain_elevation(image_path, exif_data, xmp_data, api_key):
+    """
+    Get relative terrain elevation for the image from google's terrain api.
+
+    Relative terrain elevation is the flight home point terrain elevation - the image terrain elevation.
+    The relative altitude stored in an image's xmp data is relative to the home point takeoff, not the
+    altitude of the drone above the ground at each image location. We add the relative terrain elevation
+    to the relative altitude to get the actual altitude of the drone above the ground.
+    """
+    home_lat, home_lon = get_home_point(image_path, exif_data, xmp_data)
+    image_lat, image_lon = get_lat_lon(image_path, exif_data)
+    params = {
+        "locations": f"{home_lat} {home_lon} | {image_lat} {image_lon}",
+        "key": api_key,
+    }
+    response = requests.request("GET", TERRAIN_URL, params=params).json()
+    if response["status"] != "OK":
+        logger.warning("Couldn't access google terrain api")
+        raise ParsingError("Couldn't access google terrain api")
+    home_elevation = response["results"][0]["elevation"]
```

Comment:
The one downside I can see here is that if you're getting terrain altitude for a full field of imagery (which will probably be the most common use case), you're making a ton of redundant API calls for the flight's home point. Do you see that becoming an issue?

Reply:
I have no idea. I'll add the ability for the user to optionally pass the home point altitude

Reply:
Although there is no guarantee images are from the same flight, so the home point could change. Parsing it each time seems like the safest option, and if we run into too many calls, we can update. So I'll leave things how they are now

Reply:
Or we could memoize the home point altitude, so if the home point matches, we don't make the call. That seems like the best solution